src/pages/Lend/LendByCategoryPage.vue

```diff
@@ -33,17 +31,23 @@
 import _get from 'lodash/get';
 import _map from 'lodash/map';
 import lendByCategoryQuery from '@/graphql/query/lendByCategory.graphql';
-import categoriesByIdQuery from '@/graphql/query/categoriesById.graphql';
+import loanChannelQuery from '@/graphql/query/loanChannelData.graphql';
 import WwwPage from '@/components/WwwFrame/WwwPage';
 import CategoryRow from '@/components/LoansByCategory/CategoryRow';
-import CategoryAdminControls from './CategoryAdminControls';
-import LoadingOverlay from './LoadingOverlay';
+
+// Parse a SettingsManager value
+function readSetting(data, key) {
+	try {
+		return JSON.parse(JSON.parse(_get(data, key)));
+	} catch (e) {
+		return null;
+	}
+}
 
 export default {
 	components: {
-		CategoryAdminControls,
+		CategoryAdminControls: () => import('./admin/CategoryAdminControls'),
```

Comment:
just curious about the benefit of importing this here rather than above?

Reply:
This turns it into an async component, which means it and its dependencies won't be fetched until the component is used in the page. So if a user is not an admin, we avoid even downloading the js for the admin component.